githubbot/githubbot/http.go

```diff
@@ -52,19 +54,73 @@ func (h *HTTPSrv) handleWebhook(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	var message string
+	type genericPayload interface {
+		GetRepo() *github.Repository
+	}
 	var repo string
-	branch := "master"
 	switch event := event.(type) {
-	case *github.IssuesEvent:
-		author := getPossibleKBUser(h.kbc, h.db, h.DebugOutput, event.GetSender().GetLogin())
-		message = formatIssueMsg(event, author.String())
+	case *github.PushEvent:
+		// PushEvent.GetRepo() returns *github.PushEventRepository instead of *github.Repository, so we handle it separately
 		repo = event.GetRepo().GetFullName()
-		branch, err = getDefaultBranch(repo, github.NewClient(nil))
+	default:
+		if evt, ok := event.(genericPayload); ok {
+			repo = evt.GetRepo().GetFullName()
+		} else {
+			h.Debug("could not get repo from webhook, webhook type: %s\n", github.WebHookType(r))
```

Comment:
Maybe we should have a debug channel where these messages go to, otherwise they are kind of lost in the wind

Reply:
hmm, might be worth adding into the base package like announcements. i can make a ticket